src/lib/tabs/tab-body.ts

```diff
@@ -202,4 +210,29 @@ export class MatTabBody implements OnInit {
         position == 'left-origin-center' ||
         position == 'right-origin-center';
   }
+
+  /** Computes the position state that will be used for the tab-body animation trigger. */
+  private _computePositionAnimationState(dir: Direction = this._getLayoutDirection()) {
+    if (this._positionIndex < 0) {
+      this._position = dir == 'ltr' ? 'left' : 'right';
+    } else if (this._positionIndex > 0) {
+      this._position = dir == 'ltr' ? 'right' : 'left';
+    } else {
+      this._position = 'center';
+    }
+  }
+
+  /**
+   * Computes the position state based on the specified origin position. This is used if the
+   * tab is becoming visible immediately after creation.
+   */
+  private _computePositionFromOrigin(): MatTabBodyPositionState {
+    const dir = this._getLayoutDirection();
+
+    if ((dir == 'ltr' && this.origin <= 0) || (dir == 'rtl' && this.origin > 0)) {
+      return 'left-origin-center';
+    } else {
```

Comment:
The `else` here is redundant, you can just return at the bottom.

Reply:
Just took that logic from above, but makes sense. Will improve it.